compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -289,19 +289,29 @@ pub fn write_mir_pretty<'tcx>(
             }
             Ok(())
         };
-        match tcx.hir().body_const_context(def_id.expect_local()) {
-            None => render_body(w, tcx.optimized_mir(def_id))?,
-            // For `const fn` we want to render the optimized MIR. If you want the mir used in
-            // ctfe, you can dump the MIR after the `Deaggregator` optimization pass.
-            Some(rustc_hir::ConstContext::ConstFn) => {
-                render_body(w, tcx.optimized_mir(def_id))?;
-                writeln!(w)?;
-                writeln!(w, "// MIR FOR CTFE")?;
-                // Do not use `render_body`, as that would render the promoteds again, but these
-                // are shared between mir_for_ctfe and optimized_mir
-                write_mir_fn(tcx, tcx.mir_for_ctfe(def_id), &mut |_, _| Ok(()), w)?;
+
+        match def_id.as_local() {
```

Comment:
Hmm this would work mostly (we won't see any more mir for ctfe for `const fn` though).

What my comment in https://github.com/rust-lang/rust/issues/81200#issuecomment-763500783 was referring to was to not make any difference between local MIR and MIR from other crates. So you can invoke `tcx.is_const_fn_raw(def_id)`. If it returns `true`, then you run the code in the `Some(rustc_hir::ConstContext::ConstFn)` branch below, if it is `false`, you run the code you have now added (using `instance_mir`.

The one thing I don't know is whether we should attempt to use `try_lookup` instead of `unknown` if the `DefId` is local. That could potentially resolve the other failures you're seeing.